### PR TITLE
add updateOpenApi jyy

### PR DESCRIPTION
**Pull request by jinyangyang222**

### CODE REVIEW

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java [hunk 1]
```diff
@@ -6,229 +6,246 @@
...
+        String propertiesStr = null;
+        String requiredStr = null;
+        try {
+            propertiesStr =
+                    mapper.writeValueAsString(
+                            deployVariableValidator.getVariableApiInfoMap(deployVariables)).replace(
+                            "\"[", "[").replace("]\"", "]").replace("\\", "").replace("'", "");
+            requiredStr = mapper.writeValueAsString(
+                    deployVariableValidator.getRequiredKeySet(deployVariables));
+        } catch (JsonProcessingException e) {
```
Comment (baixinsui) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java:73]: We should use log to out put the error info

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java [hunk 2]
```diff
@@ -257,7 +274,8 @@ public static String updateServiceApi(RegisterServiceEntity registerService) {
      *
      * @param id ID of registered service.
      */
-    public static void deleteServiceApi(String id) {
+    @Async("taskExecutor")
```
Comment (baixinsui) [inline modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java:277]: @Async is not need here.

File: modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java [hunk 3]
```diff
@@ -129,6 +129,7 @@ public Response update(
         String successMsg = String.format(
                 "Update registered service with id %s success.", id);
         log.info(successMsg);
+        this.orchestratorService.updateOpenApi(id);
```
[the remark this answers is not in the document]
Comment (jinyangyang222) [reply modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java:132]: ok, @async i added, the mothod updateOpenApi is 
asynchronous now.

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/OrchestratorService.java [hunk 4]
```diff
@@ -378,7 +380,38 @@ public String getOpenApiUrl(String id) {
...
         }
         // TODO find the path of swagger-ui.html of the registered by id or generate swagger-ui.html
-        return null;
+        String rootPath = System.getProperty("user.dir");
+        File folder = new File(rootPath + "/openapi");
+        File file = new File(folder, uuid + ".html");
+        if (file.exists()) {
+            return "http://localhost:8080/openapi/" + uuid + ".html";
+        } else {
+           return OpenApiUtil.updateServiceApi(registerService);
```
[the remark this answers is not in the document]
Comment (jinyangyang222) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/OrchestratorService.java]: There method updateServiceApi is disassembled updateServiceApi and updateServiceApi , 
no need to do this

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        String rootPath = System.getProperty("user.dir");
+        File folder = new File(rootPath + "/openapi");
+        File file = new File(folder, registerService.getId() + ".html");
+        if (file.exists()) {
+            file.delete();
+        }
+        List<DeployVariable> context = registerService.getOcl().getDeployment().getContext();
+        Set<String> requiredSet = new HashSet<>();
+        Map<String, Map<String, String>> propertiesMap = new HashMap<>();
+
```
[the remark this answers is not in the document]
Comment (jinyangyang222) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java]: ok,this already revise

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java (hunk 1 above)
Comment (jinyangyang222) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java:73]: ok,i will update

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java (hunk 2 above)
Comment (jinyangyang222) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java:277]: ok, fit to OrchestratorService clss the method

**Pull request merged**